manifests/redhat.pp

```diff
@@ -1,52 +1,71 @@
-# Class: datadog_agent::redhat::agent6
+# Class: datadog_agent::redhat
 #
 # This class contains the DataDog agent installation mechanism for Red Hat derivatives
 #
 
-class datadog_agent::redhat::agent6(
-  String $baseurl = $datadog_agent::params::agent6_default_repo,
-  String $gpgkey = 'https://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public',
+class datadog_agent::redhat(
+  Integer $agent_major_version = $datadog_agent::params::agent_major_version,
+  Optional[String] $agent_repo_uri = undef,
   Boolean $manage_repo = true,
   String $agent_version = $datadog_agent::params::agent_version,
   String $service_ensure = 'running',
   Boolean $service_enable = true,
   Optional[String] $service_provider = undef,
 ) inherits datadog_agent::params {
 
-  validate_legacy('Boolean', 'validate_bool', $manage_repo)
-  validate_legacy('Boolean', 'validate_bool', $service_enable)
   if $manage_repo {
-    $public_key_local = '/etc/pki/rpm-gpg/DATADOG_RPM_KEY.public'
 
-    validate_legacy('String', 'validate_string', $baseurl)
+    case $agent_major_version {
+      5 : {
+        $defaulturl = "https://yum.datadoghq.com/rpm/${::architecture}/"
+        $gpgkey = 'https://yum.datadoghq.com/DATADOG_RPM_KEY.public'
+      }
+      6 : {
+        $defaulturl = "https://yum.datadoghq.com/stable/6/${::architecture}/"
+        $gpgkey = 'https://yum.datadoghq.com/DATADOG_RPM_KEY.public'
+      }
+      7 : {
+        $defaulturl = "https://yum.datadoghq.com/stable/7/${::architecture}/"
+        $gpgkey = 'https://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public'
+      }
+      default: { fail('invalid agent_major_version') }
+    }
+
+    if ($agent_repo_uri != undef) {
+      $baseurl = $agent_repo_uri
+    } else {
+      $baseurl = $defaulturl
+    }
 
-    file { 'DATADOG_RPM_KEY.public':
+    $public_key_local = '/etc/pki/rpm-gpg/DATADOG_RPM_KEY.public'
+
+    file { 'DATADOG_RPM_KEY_E09422B3.public':
         owner  => root,
         group  => root,
         mode   => '0600',
         path   => $public_key_local,
-        source => $gpgkey
+        source => 'https://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public'
     }
 
     exec { 'install-gpg-key':
         command => "/bin/rpm --import ${public_key_local}",
         onlyif  => "/usr/bin/gpg --dry-run --quiet --with-fingerprint -n ${public_key_local} | grep 'A4C0 B90D 7443 CF6E 4E8A  A341 F106 8E14 E094 22B3' || gpg --dry-run --import --import-options import-show ${public_key_local} | grep 'A4C0B90D7443CF6E4E8AA341F1068E14E09422B3'",
         unless  => '/bin/rpm -q gpg-pubkey-e09422b3',
-        require => File['DATADOG_RPM_KEY.public'],
+        require => File['DATADOG_RPM_KEY_E09422B3.public'],
     }
 
-    yumrepo {'datadog':
+    yumrepo {'datadog5':
```

Comment:
OK, I see we are ensuring these are absent, I think we might also need to ensure the `datadog-beta` repo is absent, as it was used by the `facter`. It's not 100% clear to me who'd be affected but I _think_ there might be the weird case where we'd want to clean that up.